card_game.py

```diff
@@ -0,0 +1,214 @@
+        self.name = name
+        self.max_hp = max_hp
+        self.current_hp = max_hp
+        self.ap = ap
+        self.deck = deck if deck else []
+        self.max_hand = 5
+        self.hand_card = []
+        self.used_card = []
+        self.defense_point = 0
+
+    def take_damage(self, damage):
+        if self.defense_point > 0:
+            damage_after_defense = max(0, damage - self.defense_point)
+            self.defense_point = max(0, self.defense_point - damage)
+            self.current_hp -= damage_after_defense
+        else:
+            self.current_hp -= damage
+
+    def add_defense(self, defense_point):
+        self.defense_point += defense_point
+
+    def draw_additional_cards(self, draw_count):
+        random.shuffle(self.deck)
+        additional_cards = self.deck[:draw_count]
+        self.hand_card.extend(additional_cards)
+
+    def play_card(self, card, target):
+        if card.ap <= self.ap:
+            card.play(self, target)
+            self.ap -= card.ap
+            self.hand_card.remove(card)
+            self.used_card.append(card)
+        else:
+            print("Not enough action points to play this card.")
+
+
+class Player(Character):
+    def display_card_list(self):
+        print("Your hand:")
+        for i, card in enumerate(self.hand_card, 1):
+            print(f"{i}. {card.name} (AP: {card.ap}) - {card.info}")
+
+
+class Enemy(Character):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.ap = 0  # Enemies have 0 action points
+        self.max_hand = 1
+
+
+class Game:
+    def __init__(self):
+        self.player = None
+        self.enemy = None
+
+    def create_characters(self):
+        # Initialize player and enemy characters
+        # The line below is removed since we initialize player and enemy later in the main function
+        # self.player = Player("Player", 30, deck=[Card(...), ...])  # Add player's cards
+        # self.enemy = Enemy("Enemy", 20, deck=[Card(...), ...])  # Add enemy's cards
+        pass
+
+    def display_info(self):
+        print("\n" + "-" * 30)
+        print(
+            f"Player HP: {self.player.current_hp}. Defense: {self.player.defense_point}")
+        print(
+            f"Enemy HP: {self.enemy.current_hp}. Defense: {self.enemy.defense_point}")
+        print("Enemy Intent:", self.enemy.hand_card[0].info)
+
+    def check_game_status(self):
+        if self.player.current_hp <= 0:
+            print("Game Over. You lost.")
+            return False
+
+        if self.enemy.current_hp <= 0:
+            print("You WIN!!")
+            return False
+
+        return True
+
+    def player_turn(self):
+        # Move remaining cards in hand to used_card
+        self.player.used_card.extend(self.player.hand_card)
+        self.player.hand_card.clear()
+
+        self.player.draw_additional_cards(self.player.max_hand)
+        self.player.ap = 3
```

Comment:
nit: create a constant variable like `PLAYER_AP = 3`, and replace all of these hard-code 3 with the variable, e.g. this line and `Character().__init__()`.

Reply:
where should I put my constant variable?

Reply:
Added ap_max in the Character class.